### PR TITLE
Fix MetricsSpec

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala [hunk 1]
```diff
@@ -24,52 +23,31 @@ object TestReporter {
...
+  def testMetrics[T<:Metric](metricNamePattern: String)(implicit scenarioName: ProcessName): Iterable[T] =
+    processToMetric.asScala.filterKeys { case (scenarioName1, metricName1) =>
+      scenarioName1 == scenarioName && metricName1.contains(metricNamePattern)
+    }.values.map(_.asInstanceOf[T])
 
-  def testHistogram(containing: String): Histogram = {
-    testHistograms.filter(_._2.contains(containing)).keys.head
+  override def notifyOfAddedMetric(metric: Metric, metricName: String, group: MetricGroup): Unit = {
+    val metricId = group.getMetricIdentifier(metricName, this)
+    processToMetric.put((ProcessName(group.getAllVariables.get("<job_name>")), metricId), metric)
```
Comment (arkadius) [inline engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala:48]: Can you use some constant or add notice why u know that it is in job_name?

File: engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala [hunk 2]
```diff
@@ -24,52 +23,31 @@ object TestReporter {
...
-    gauges.clear()
-    counters.clear()
-    meters.clear()
-  }
+  private val processToMetric = new ConcurrentHashMap[(ProcessName, String), Metric]()
 
-  def testHistograms: Map[Histogram, String] = histograms.asScala.toMap
+  def testMetrics[T<:Metric](metricNamePattern: String)(implicit scenarioName: ProcessName): Iterable[T] =
+    processToMetric.asScala.filterKeys { case (scenarioName1, metricName1) =>
+      scenarioName1 == scenarioName && metricName1.contains(metricNamePattern)
```
Comment (arkadius) [inline engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala:43]: So after looking in the code, I see that there was two problem:
- we returned metrics for other concurrent scenarios
- we tested instant rate 
Have u reproduced this bugs?

**Review by arkadius — COMMENTED** [review]
So

File: engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala]: :+1: I found in Flink  :)

**Review by arkadius — APPROVED** [review]

File: engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala:44]: TBH, I tried mulitple times locally and I haven't succeeded... We have to try it out on CI...

File: engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala [hunk 3]
```diff
@@ -24,52 +23,31 @@ object TestReporter {
...
-    ).getOrElse(
-      throw new IllegalArgumentException("Reporter doesn't exists.")
-    )
+  def get(name: String): TestReporter = TestReporter.instances.asScala
+    .getOrElse(name, throw new IllegalArgumentException("Reporter doesn't exists."))
+
 }
 
-class TestReporter extends AbstractReporter {
+class TestReporter extends MetricReporter with CharacterFilter {
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala:38]: I don't think it makes sense to mention those... this is quite specific test

[the remark this answers is not in the document]
Comment (mproch) [reply engine/flink/test-utils/src/main/scala/pl/touk/nussknacker/engine/process/functional/TestReporter.scala:38]: :+1: You were right, I checked and it was used in one place, so I added entry in migration guide...

**Pull request merged**